core/src/main/java/org/apache/accumulo/core/conf/Property.java

```diff
@@ -429,15 +429,41 @@ public enum Property {
   MANAGER_FATE_METRICS_MIN_UPDATE_INTERVAL("manager.fate.metrics.min.update.interval", "60s",
       PropertyType.TIMEDURATION, "Limit calls from metric sinks to zookeeper to update interval.",
       "1.9.3"),
-  MANAGER_FATE_THREADPOOL_SIZE("manager.fate.threadpool.size", "64", PropertyType.COUNT,
-      "The number of threads used to run fault-tolerant executions (FATE)."
-          + " These are primarily table operations like merge.",
+  @Deprecated(since = "4.0.0")
+  MANAGER_FATE_THREADPOOL_SIZE("manager.fate.threadpool.size", "64",
+      PropertyType.FATE_THREADPOOL_SIZE,
+      "Previously, the number of threads used to run fault-tolerant executions (FATE)."
+          + " This is no longer used in 4.0+. MANAGER_FATE_USER_CONFIG and"
+          + " MANAGER_FATE_META_CONFIG are the replacement and must be set instead.",
       "1.4.3"),
+  MANAGER_FATE_USER_CONFIG("manager.fate.user.config", "{"
+      + "\"TABLE_CREATE,TABLE_DELETE,TABLE_RENAME,TABLE_ONLINE,TABLE_OFFLINE,NAMESPACE_CREATE,NAMESPACE_DELETE,NAMESPACE_RENAME,TABLE_TABLET_AVAILABILITY,SHUTDOWN_TSERVER\": 1,"
+      + "\"TABLE_BULK_IMPORT2\": 2,"
+      + "\"TABLE_COMPACT,TABLE_CANCEL_COMPACT,COMMIT_COMPACTION\": 4,"
```

Comment:
COMMIT_COMPACTION will likely be the most frequently executed fate operation in a system.  A single TABLE_COMPACT operation can cause N COMMIT_COMPACTION operations to eventually run.  Also system compactions that occur automatically as data is ingested can cause COMMIT_COMPACTION ops to run.  Should probably move TABLE_COMPACT,TABLE_CANCEL_COMPACT elsewhere w/ the other table ops as they are much less frequent operations and are always user initiated.

I would recommend three default groups.   One for COMMIT_COMPACTION, one for SYSTEM_SPLIT, and one for everythnig else (I think everything else is the user initiated operations).  This will give dedicated resrouces to the the two system initated operations and keep them from intefering w/ the user initiated ops.